metrics-api/src/main/java/software/amazon/swage/metrics/MetricContext.java

```diff
@@ -0,0 +1,102 @@
+package software.amazon.swage.metrics;
+
+import java.io.Closeable;
+import java.time.Instant;
+
+import software.amazon.swage.collection.TypedMap;
+
+/**
+ * The context in which a measurement is taken. A {MetricContext} contains the parameters of
+ * the environment in which a measurement is taken.
+ * In the physical world, these might define the effective time and place;
+ * in a computer system, they might define the host, request, function, task or
+ * other application-specific parameters.
+ * <p/>
+ * A {MetricContext} is the primary interface used to record measurements.
+ */
+// TODO: provide a childContext() mechanism?
+public final class MetricContext implements Closeable {
+    private final MetricRecorder recorder;
+    private final TypedMap dimensions;
+
+    /**
+     * Create a new context for taking measurements.
+     *
+     * @param recorder   where measurments will be recorded
+     * @param dimensions the properties of this context
+     */
+    public MetricContext(MetricRecorder recorder, TypedMap dimensions) {
+        this.recorder = recorder;
+        this.dimensions = dimensions;
+    }
+
+    /**
+     * Returns dimensions to associate with
+     *
+     * @return the Context dimensions
+     */
+    public TypedMap dimensions() {
```

Comment:
There is no guarantee that the TypedMap is immutable, so this may expose mutable state.  Shouldn't, but may.

Reply:
I don't think TypedMap has any mutating operations.